pkg/config/config.go

```diff
@@ -45,6 +45,23 @@ type OIDC struct {
 	FriendlyNameClaimName  string            `mapstructure:"friendly_name_claim_name"`
 }
 
+// TODO(isan) consolidate with OIDC
+// CookieAuthVerification is related to auth based on a cookie set by an external service
+type CookieAuthVerification struct {
+	// ValidateIDTokenClaims if set will validate the values  (e.g department: "R&D") exist in the token claims
+	ValidateIDTokenClaims map[string]string `mapstructure:"validate_id_token_claims"`
```

Comment:
Why do we need the `mapstructure` everywhere?

Reply:
,We have this in 99% of our configuration fields, it keeps persistent interface for the configuration even if the field name changes. Not sure if I understood 100% your question :)